tests/testthat/test-xctr.R

```diff
@@ -282,3 +282,23 @@ test_that("for a company with 3 products of varying footprints, value is 1/3 (#2
   out <- xctr(companies, co2, low_threshold, high_threshold)
   expect_true(identical(unique(out$value), expected_value))
 })
+
+test_that("isic sector column has character class", {
+  companies <- tibble(
+    company_id = rep("a"),
+    clustered = c("b"),
+    activity_uuid_product_uuid = c("c"),
+  )
+  co2 <- tibble(
+    activity_uuid_product_uuid = c("c"),
+    co2_footprint = 1,
+    tilt_sector = "transport",
+    unit = "metric ton*km",
+    isic_4digit = "4575",
+  )
+  out <- co2 |>
+    xctr_rename() |>
+    xctr_add_ranks(col_to_rank(co2)) |>
+    xctr_add_scores(low_threshold = 1/3, high_threshold = 2/3)
+  expect_true(is.character(unlist(select(out ,contains("isic_sec")))))
+})
```

Comment:
I think the test we need is one that excercises the new `stop_if*()` function you created. 

The tests you need are easier to see when you write the test before the code (test-driven-development TDD). It would be something similar to this:

```r
test_that("if the 'isic' column is a character, throws the expected error", {
  ...
  
  expect_error(xctr(companies, products (or inputs), "a piece of the error message you expect")
})
```

Also note you always test the user-facing function, i.e. `xctr*()`. If you test the internal functions your tests will be brittle and will fail when you change the internal implementation.

Reply:
@maurolepore Don't you think this type of test is already part of other tests present in the file? Btw I happy to do the change as you suggested if you think otherwise

Reply:
@maurolepore I will make tests keeping that suggestion in mind in the future. thanks!